relayer/pkg/chainlink/txm/txm.go

```diff
@@ -302,13 +370,42 @@ func (txm *starktxm) confirmLoop() {
 						continue
 					}
 					response, err := client.Provider.GetTransactionStatus(ctx, f)
+
+					// tx can be rejected due to a nonce error. but we cannot know from the Starknet RPC directly  so we have to wait for
+					// a broadcasted tx to fail in order to fix the nonce errors
+
 					if err != nil {
 						txm.lggr.Errorw("failed to fetch transaction status", "hash", hash, "error", err)
 						continue
 					}
 
 					status := response.FinalityStatus
 
+					// currently, feeder client is only way to get rejected reason
+					if status == starknetrpc.TxnStatus_Rejected {
+						feederClient, err := txm.feederClient.Get()
+						if err != nil {
+							txm.lggr.Errorw("failed to load feeder client", "error", err)
+							break
+						}
+
+						rejectedTx, err := feederClient.TransactionFailure(ctx, f)
+						if err != nil {
+							txm.lggr.Errorw("failed to fetch reason for transaction failure", "hash", hash, err)
+							continue
+						}
+
+						if isInvalidNonce(rejectedTx.ErrorMessage) {
```

Comment:
nope! at least not for now. i think in the future we can think about rebroadcasting. but part of having the feeder client gives us the ability to log and analyze what errors we can mitigate in the future

Reply:
that's a good question. originally, this problem had a ticket for it because starknet made some changes to tx life cycle management and as a result the txmgr could enter a state with an out of sync nonce.

i guess there is a potential optional where we just resync the nonce manager without rebroadcasting. basically, resync but don't retry txs. it would be a heck of a lot simpler. but the counterargument is we already have the solution to rebroadcast for nonces.

so tradeoff is simplicity with less robustness over complexity with robustness. 

i think it's worth discussing @archseer @cfal @aalu1418

Reply:
i would hesitate to allow the txm to just "drop" a tx - that behavior has only been limited to solana data feeds because there are no nonces + continuous/consistent updates + broadcasting schedule

typical data feeds only report on deviation changes so that makes broadcasting individuals reports is more critical since there may not be another for some time. also DONs usually only have 1 node broadcast at a time which is a cost saving method assuming a single tx can reliably included on chain